src/shiver/models/corrections.py

```diff
@@ -0,0 +1,202 @@
+"""Model for Corrections tab"""
+# pylint: disable=no-name-in-module
+import time
+from typing import Tuple
+from mantid.api import mtd, AlgorithmManager, AlgorithmObserver
+from mantid.kernel import Logger
+
+logger = Logger("SHIVER")
+
+
+class CorrectionsModel:
+    """Corrections table model"""
+
+    def __init__(self) -> None:
+        self.algorithms_observers = set()  # need to add them here so they stay in scope
+        self.error_callback = None
+
+    def apply(
+        self,
+        ws_name: str,
+        detailed_balance: bool,
+        hyspec_polarizer_transmission: bool,
+        temperature: str = "",
+    ) -> None:
+        """Apply corrections"""
+        output_ws_name = f"{ws_name}_correction"
+        if detailed_balance:
+            self.apply_detailed_balance(
+                ws_name,
+                temperature,
+                output_ws_name,
+            )
+        #
+        if hyspec_polarizer_transmission:
+            input_ws_name = ws_name
+            if detailed_balance:
+                # wait for detailed balance to finish
+                while not mtd.doesExist(output_ws_name):
+                    time.sleep(0.1)
+                input_ws_name = output_ws_name
+            self.apply_scattered_transmission_correction(
+                input_ws_name,
+                output_ws_name,
+            )
+
+    def apply_detailed_balance(
+        self,
+        ws_name: str,
+        temperature: str,
+        output_ws_name: str,
+    ) -> None:
+        """Apply detailed balance to the workspace"""
+        alg_obs = ApplyDetailedBalanceMDObserver(
+            parent=self,
+            ws_name=ws_name,
+        )
+        self.algorithms_observers.add(alg_obs)
+
+        alg = AlgorithmManager.create("ApplyDetailedBalanceMD")
+
+        alg_obs.observeFinish(alg)
+        alg_obs.observeError(alg)
+
+        alg.initialize()
+        alg.setLogging(False)
+        try:
+            alg.setProperty("InputWorkspace", ws_name)
+            alg.setProperty("Temperature", temperature)
+            alg.setProperty("OutputWorkspace", output_ws_name)
+            alg.executeAsync()
+        except ValueError as err:
+            logger.error(str(err))
+            if self.error_callback:
+                self.error_callback(str(err))
+
+    def apply_scattered_transmission_correction(
+        self,
+        ws_name: str,
+        output_ws_name: str,
+    ) -> None:
+        """Apply DGSScatterTransmissionCorrection to the workspace"""
+        alg_obs = DgsScatteredTransmissionCorrectionMDObserver(
+            parent=self,
+            ws_name=ws_name,
+        )
+        self.algorithms_observers.add(alg_obs)
+
+        exponent_factor = 1.0 / 11.0  # fixed value
```

Comment:
IOP Conf. Series: Journal of Physics: Conf. Series 862 (2017) 012023 doi:10.1088/1742-6596/862/1/012023, Figure 3 right